### PR TITLE
Feature: Placeholder for the stats page tabs

### PR DESCRIPTION
**Pull request by JakubFrackowiak**
Add working buttons that open different stats components.

### CODE REVIEW

File: components/ExperienceAndRewiews.tsx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export default function ExperienceAndRewiews(): JSX.Element {
```
Comment (piotrostr) [inline components/ExperienceAndRewiews.tsx:1]: *Reviews

File: components/FreelancerBids.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { collection, query, where } from "firebase/firestore"
+import { useFirestore, useFirestoreCollectionData, useUser } from "reactfire"
+
+export default function FreelancerBids(path): JSX.Element {
+  const firestore = useFirestore()
+  const { status, data: user } = useUser()
+  const bidsRef = collection(firestore, "quests/path.path/bids")
```
Comment (piotrostr) [inline components/FreelancerBids.tsx:7]: Did you mean `${path.path}`?

File: components/Stats.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { useState } from "react"
+import QuestsStats from "components/QuestsStats"
+import ProfileStats from "components/ProfileStats"
+import ExperienceAndRewiews from "components/ExperienceAndRewiews"
+
+export default function Stats(): JSX.Element {
+  const [showStats, setShowStats] = useState("Quests")
+  const tabs = ["Quests", "Profile", "Epirience And Reviews"]
```
Comment (piotrostr) [inline components/Stats.tsx:8]: *Experience

File: components/Stats.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  function Tab(tabName): JSX.Element {
```
Comment (piotrostr) [inline components/Stats.tsx:10]: Specify type of `tabName`

File: components/Stats.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    return (
+      <div style={{ cursor: "pointer" }}>
+        <a
+          onClick={() => {
+            setShowStats(tabName.tabName)
```
Comment (piotrostr) [inline components/Stats.tsx:15]: Unclear name, don't add object keys of the same name as the object variable

How about making it `tab.name`?

File: pages/stats.tsx [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import styled from "styled-components"
 import styles from "../styles/Home.module.css"
+import Stats from "components/Stats"
```
Comment (piotrostr) [inline pages/stats.tsx:3]: Don't do default imports/exports for components

File: storage/quest/populate.ts [hunk 7]
```diff
@@ -1,17 +1,26 @@
 import { Firestore, collection, getDocs, addDoc } from "firebase/firestore"
 import { Quest } from "../../storage/quest"
 import { faker } from "@faker-js/faker"
+import { userAgent } from "next/server"
+import { getAuth } from "firebase/auth"
```
Comment (piotrostr) [inline storage/quest/populate.ts:6]: Unstage the `populate.ts` file as it has changed, alternatively merge the current `main` branch into `stats-page-tabs` branch

File: storage/quest/types.ts [hunk 8]
```diff
@@ -18,6 +18,8 @@ export interface Quest {
   tags: Tag[]
   bids?: Bid[]
   id?: string
```
Comment (piotrostr) [inline storage/quest/types.ts:20]: Same, this file has also changed

**Review by piotrostr — CHANGES_REQUESTED** [review]

File: components/AllStats.tsx [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  const biddingQuestsQuery = query(
+    questsRef,
+    where("bidders", "array-contains", user?.uid || "")
+  )
+
+  const { status: biddingQuestsStatus, data: biddingQuests } =
+    useFirestoreCollectionData(biddingQuestsQuery)
+  return (
+    <div>
+      {quests?.map((quest, idx) => (
```
Comment (OliwierOST) [inline components/AllStats.tsx:23]: Quest should have a type

File: components/AllStats.tsx [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        <div key={idx}>
+          <div>{quest?.title}</div>
+          <div>{quest?.description}</div>
+          <div>{quest?.reward}</div>
+          <div>{quest?.tags}</div>
+          <div>{quest?.userId}</div>
+        </div>
+      ))}
+      {biddingQuests?.map((biddingQuest, idx) => (
```
Comment (OliwierOST) [inline components/AllStats.tsx:32]: BiddingQuest as well should have a type

File: components/Stats.tsx [hunk 11]
```diff
@@ -0,0 +1,37 @@
+import { useState } from "react"
+import {
+  ProfileStats,
+  ExperienceAndReviews,
+  QuestsStats,
+} from "components/AllStats"
+
+export function Stats(): JSX.Element {
+  const [showStats, setShowStats] = useState("Quests")
+  const tabs = ["Quests", "Profile", "Experience And Reviews"]
```
Comment (OliwierOST) [inline components/Stats.tsx:10]: Please instead of mapping through tabs array make separate component for each tab.

File: components/Stats.tsx [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+          }}
+        >
+          <div>Show {tab.name}</div>
+        </a>
+      </div>
+    )
+  }
+  return (
+    <div>
+      <div>
```
Comment (OliwierOST) [inline components/Stats.tsx:27]: This div could be an empty tag

**Review by OliwierOST — CHANGES_REQUESTED** [review]

File: components/Stats.tsx (hunk 11 above)
Comment (OliwierOST) [reply components/Stats.tsx:10]: You could make a folder for all components for tabs

Comment (piotrostr) [reply components/Stats.tsx:10]: **Should**, we don't want a flood in the components folder lol

File: components/AllStats.tsx [hunk 13]
```diff
@@ -37,3 +37,30 @@ export function QuestsStats(): JSX.Element {
...
+  const { status, data: user } = useUser()
+  const bidsRef = collection(firestore, "quests", `${path.path}`, "bids")
+  const bidsQuery = query(bidsRef, where("userId", "==", user?.uid || ""))
+  const { status: bidsStatus, data: bids } =
+    useFirestoreCollectionData(bidsQuery)
+  return (
+    <div>
+      {bids?.map((bid, idx) => (
+        <div key={idx}>
+          <div>{bid?.amount}</div>
```
Comment (piotrostr) [inline components/AllStats.tsx:52]: This should be mapping into `<Bid />` components

**Review by piotrostr — CHANGES_REQUESTED** [review]

File: components/Tabs/QuestStats.tsx [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+              </>
+            )}
+          </>
+        )}
+      </>
+    </div>
+  )
+}
+
+export function FreelancerBids(path): JSX.Element {
```
[the remark this answers is not in the document]
Comment (OliwierOST) [reply components/Tabs/QuestStats.tsx:75]: Also what type does it have?

**Review by OliwierOST — CHANGES_REQUESTED** [review]

File: components/Tabs/QuestStats.tsx [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+      </>
+      <>
+        {biddingQuestsStatus && (
+          <>
+            {biddingQuestsStatus === "loading" ? (
+              <div>loading bids</div>
+            ) : (
+              <>
+                {biddingQuests ? (
+                  biddingQuests?.map((biddingQuest: Bid, idx) => (
```
Comment (OliwierOST) [inline components/Tabs/QuestStats.tsx:58]: Notice how confusing the name biddingQuest is. You can see that it's actually a Bid by it's type. I would suggest changing it to something like userBids or currentUserBids.

**Review by OliwierOST — CHANGES_REQUESTED** [review]

**Comment by OliwierOST** [comment]
There are still some unresolved changes @JakubFrackowiak, please have another look

If it is a false-positive, please mark the outstanding unresolved comments and get me a green light! :)

As of now, merging is still blocked

File: components/Stats.tsx [hunk 16]
```diff
@@ -0,0 +1,55 @@
+import { useState } from "react"
+import { QuestsStats } from "components/Tabs/QuestStats"
```
Comment (OliwierOST) [inline components/Stats.tsx:2]: There should be no absolute imports

Please make changes everywhere this occurs

File: components/Stats.tsx [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+import styled from "styled-components"
+
+const Button = styled.div`
+  cursor: pointer;
+`
+
+export function Stats(): JSX.Element {
+  const [showStats, setShowStats] = useState("QuestsStats")
+
+  function Tabs(tab): JSX.Element {
```
Comment (OliwierOST) [inline components/Stats.tsx:14]: The tab should have a type specified in `interface Props {...}`

File: components/Stats.tsx [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+    return (
+      <>
+        <Button>
```
Comment (OliwierOST) [inline components/Stats.tsx:17]: We are using the MUI components library for the project

Please use them here wherever you can for consistency

File: components/Tabs/ExpirienceAndReviews.tsx.tsx [hunk 19]
```diff
@@ -0,0 +1,3 @@
+export function ExperienceAndReviews(): JSX.Element {
```
Comment (OliwierOST) [inline components/Tabs/ExpirienceAndReviews.tsx.tsx:1]: What is the intended purpose for this component?

File: components/Tabs/QuestStats.tsx [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+              </>
+            )}
+          </>
+        )}
+      </>
+    </div>
+  )
+}
+
+export function FreelancerBids({ path }): JSX.Element {
```
Comment (OliwierOST) [inline components/Tabs/QuestStats.tsx:75]: The prop `path` should be given a type

**Review by OliwierOST — CHANGES_REQUESTED** [review]

File: components/Tabs/ExpirienceAndReviews.tsx.tsx (hunk 19 above)
Comment (JakubFrackowiak) [reply components/Tabs/ExpirienceAndReviews.tsx.tsx:1]: Tab for expirience and reviews?

File: components/Tabs/QuestStats.tsx (hunk 20 above)
Comment (JakubFrackowiak) [reply components/Tabs/QuestStats.tsx:75]: component not working with type for path

**Comment by OliwierOST** [comment]
Please refactor the components and use the MUI library. We can't hold PRs for so long

**Pull request closed without merging**